src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -7437,8 +7437,12 @@ private SyntaxList<AttributeListSyntax> ParseStatementAttributeDeclarations()
 
             // Continue consuming element access expressions for `[x][y]...`.  We have to determine if this is a
             // collection expression being indexed into, or if it's a sequence of attributes.
+            var bracketArgumentListCount = 0;
```

Comment:
I didn't understand why we are counting these instead of just flipping a bit to `true`.

Reply:
good call.  will make that a bool.